tpl/crypto/crypto.go

```diff
@@ -63,3 +67,44 @@ func (ns *Namespace) SHA256(in interface{}) (string, error) {
 	hash := sha256.Sum256([]byte(conv))
 	return hex.EncodeToString(hash[:]), nil
 }
+
+// HMAC hashes the concatenation of a message and a secret key
+// and returns its checksum.
+func (ns *Namespace) HMAC(h interface{}, m interface{}, k interface{}) (string, error) {
```

Comment:
I didn't notice this when you proposed this function in the forums, but I think the function signature should have the message last.  In Go templates, if you use a pipeline, the piped value is appended to the function arguments, so always design functions to accept the main user input as the last argument.  These are equivalent:

```
{{ $message | hmac "sha256" $key }}
{{ hmac "sha256" $key $message }}
```